src/app2.js

```diff
@@ -0,0 +1,111 @@
+import axios from "axios";
+
+const countryList = document.getElementById('country-list');
+// const countryImage = document.getElementById('country-flag')
+
+//1. data fetchen
+
+//
+// const usernameInputField = document.getElementById('input-search');
+
+async function fetchDataSearch(countryName) {
+
+    // const URI = 'https://restcountries.com/v2/name/${countryName}'
+    const response = await axios.get(`https://restcountries.com/v2/name/${countryName}`)
+
+
+      try {
+
+        // console.log(response.data);
+
+        response.data.sort ((a, b) => (a.population - b.population));
+
+        //empty after search
+        countryList.replaceChildren();
+
+        response.data.map((country) => {
+            const {flags:{png}, name, capital, subregion, currencies, population, languages} = country;
+
+            //country names;
+            const itemName = document.createElement('p');
+            itemName.setAttribute('class', 'countryName');
+            itemName.textContent = name;
+
+            //flags
+            const itemPopulation = document.createElement('p');
+            itemPopulation.setAttribute('class', 'countryPopulation');
+            itemPopulation.textContent = "population of : " + population;
+
+            // sortPopulation(country.population)
+            const itemFlag = document.createElement('img');
+            itemFlag.setAttribute('class', 'countryImage');
+            itemFlag.setAttribute('src', png);
+
+            //make array list of needed names.
+            let coinString;
+            const coinArray = currencies.map((coin) => {
+                return coin.name;
+            })
+            // print strings
+            if (currencies.length === 1) {
+                coinString = `you can pay with ${coinArray[0]}`;
+            }else if(currencies.length === 2) {
+                coinString = `you can pay with ${coinArray[0]} and ${coinArray[1]}`;
+            } else {
+                coinString = `you can pay with ${coinArray[0]}, ${coinArray[1]} and ${coinArray[2]}`
+            }
```

Comment:
Dit is niet echt uitbreidbaar voor als er meer dan drie valuta's zouden zijn. Wellicht dat je de volgende keer iets met een for-loop of de [Array.join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) functie kan doen